tools/job-cache/job-cache.cpp

```diff
@@ -478,17 +712,99 @@ class Cache {
         input_dirs(db),
         output_files(db),
         transact(db),
+        matching_jobs(db),
         dir(std::move(_dir)),
         rng(wcl::xoshiro_256::get_rng_seed()) {}
 
+  // TODO: Unlike reading, we need to account for
+  // directory remappings here.
+  bool read(const FindJobRequest &find_request) {
```

Comment:
This does more tricky filesystem work but it should all be fairly cheap hopefully.